src/audio/copier/copier.c

```diff
@@ -639,16 +650,20 @@ static int copier_copy(struct comp_dev *dev)
 	case SOF_COMP_HOST:
 		if (!cd->ipc_gtw)
 			return do_endpoint_copy(dev);
-		break;
+
+		/* do nothing in the gateway copier case */
```

Comment:
is this a change in behaviour? It used to call `do_multi_endpoint_module_copy()` here before?

Reply:
logic changed.
host copy is neither multi_endpoint nor module copy, right? so directly return for host and do nothing for gtw case.

Reply:
@lyakh its not really a change in behaviour. It does the same thing in all cases, just in a simplified way

Reply:
@ranj063 I'm not very familiar with the IPC gateway flow, so I might be wrong. Before this change for the IPC gateway case `do_multi_endpoint_module_copy()` was called, then in it the `if (cd->endpoint_num && !cd->bsource_buffer)` condition would be satisfied and `do_endpoint_copy()` would be called? In it it would then detect that `.multi_endpoint_buffer` isn't set so it would call
```
		return cd->endpoint[0]->drv->ops.copy(cd->endpoint[0]);
```
?

Reply:
> @ranj063 I'm not very familiar with the IPC gateway flow, so I might be wrong. Before this change for the IPC gateway case `do_multi_endpoint_module_copy()` was called, then in it the `if (cd->endpoint_num && !cd->bsource_buffer)` condition would be satisfied and `do_endpoint_copy()` would be called? In it it would then detect that `.multi_endpoint_buffer` isn't set so it would call
> 
> ```
> 		return cd->endpoint[0]->drv->ops.copy(cd->endpoint[0]);
> ```
> 
> ?

but @lyakh in do_endpoint)copy() for the ipc_gtw case, all we do is return 0. whch is effectively what this PR does. Do avoid a roundabout way to return 0 for the IPC gateway case.

Reply:
I think previous logic should have something wrong for gtw case, it does go into multi_endpoint_copy and perform one time do_endpoint_copy(althrough it will return 0 immediately), this version will directly return 0 with gtw case.

So, conclusion is:
1. previous copier_copy code logic is redundant for gtw case.
2. both CI and QB are passed with this changes.

Reply:
> in do_endpoint)copy() for the ipc_gtw case, all we do is return 0. whch is effectively what this PR does.

@ranj063 oh, sorry, I was looking at my slightly outdated copy of copier.c, sorry